commandapi-platforms/commandapi-bukkit/commandapi-bukkit-core/src/main/java/dev/jorel/commandapi/network/BukkitCommandAPIMessenger.java

```diff
@@ -0,0 +1,57 @@
+package dev.jorel.commandapi.network;
+
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.plugin.java.JavaPlugin;
+import org.bukkit.plugin.messaging.Messenger;
+import org.bukkit.plugin.messaging.PluginMessageListener;
+import org.jetbrains.annotations.NotNull;
+
+// Based on https://www.spigotmc.org/wiki/bukkit-bungee-plugin-messaging-channel/
+/**
+ * A {@link CommandAPIMessenger} for sending and receiving messages to and from CommandAPIBukkit and {@link Player}s
+ * connected to the Bukkit server.
+ */
+public class BukkitCommandAPIMessenger extends CommandAPIMessenger<Player, Player> implements PluginMessageListener {
+	private final String channelName;
```

Comment:
I'm not entirely sure why the channel name has to be passed to this class, could it not directly inherit it from `CommandAPIProtocol.CHANNEL_NAME` in this class instead of requiring it in its constructor? Unless there is ever a reason for another channel name to be passed, I don't see why this is a useful thing to do.

Reply:
If we ever want to send packets on another channel, we could easily do that by creating another messenger object and passing it a different channel String.

But yes, currently there is no reason to have this in the constructor.